--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -267,8 +267,8 @@ target_include_directories(audio_hal PRIVATE
 	./audio_hal
 	${AML_BUILD_DIR})
 
-target_compile_options(audio_hal PRIVATE -D_GNU_SOURCE -DTV_AUDIO_OUTPUT
-	-Werror -Wno-error=discarded-qualifiers -Wno-deprecated-declarations -funwind-tables -fasynchronous-unwind-tables -fPIC -g)
+target_compile_options(audio_hal PRIVATE -D_GNU_SOURCE -DTV_AUDIO_OUTPUT
+	-Wno-error -Wno-error=discarded-qualifiers -Wno-deprecated-declarations -funwind-tables -fasynchronous-unwind-tables -fPIC -g)
 
 if(USE_DTV)
 	target_include_directories(audio_hal PRIVATE
@@ -317,6 +317,23 @@ install(TARGETS audio_hal
 install(FILES audio_hal/mixer_paths.xml
 	DESTINATION /etc)
 
+install(DIRECTORY include/hardware
+	DESTINATION include
+	FILES_MATCHING PATTERN "*.h")
+install(DIRECTORY include/system
+	DESTINATION include
+	FILES_MATCHING PATTERN "*.h")
+install(FILES include/Virtualx_v4.h
+	DESTINATION include)
+install(FILES ../files/aml_audio_config.t7.json
+	DESTINATION /etc/halaudio
+	RENAME aml_audio_config.json)
+install(FILES ../files/mixer_paths.t7.xml
+	DESTINATION /etc
+	RENAME mixer_paths.xml)
+install(FILES ../files/ms12_audio_profiles.ini
+	DESTINATION /etc/halaudio)
+
 if (CMAKE_SYSTEM_PROCESSOR MATCHES "(arm)")
 install(PROGRAMS aml_aq_hw/Amlogic_DRC_Param_Generator aml_aq_hw/Amlogic_EQ_Param_Generator
 	DESTINATION /usr/bin)
